allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -155,24 +159,36 @@ def __init__(
         self.sensor_preprocessor_graph = None
         self.actor_critic: Optional[ActorCriticModel] = None
         if self.num_samplers > 0:
+            create_model_kwargs = {}
             if self.machine_params.sensor_preprocessor_graph is not None:
-                # centralized observation set,
                 self.sensor_preprocessor_graph = self.machine_params.sensor_preprocessor_graph.to(
                     self.device
                 )
-                set_seed(self.seed)
-                self.actor_critic = cast(
-                    ActorCriticModel,
-                    self.config.create_model(
-                        sensor_preprocessor_graph=self.sensor_preprocessor_graph
-                    ),
-                ).to(self.device)
-            else:
-                # no observation set
-                set_seed(self.seed)
-                self.actor_critic = cast(
-                    ActorCriticModel, self.config.create_model()
-                ).to(self.device)
+                create_model_kwargs[
+                    "sensor_preprocessor_graph"
+                ] = self.sensor_preprocessor_graph
+
+            set_seed(self.seed)
+            self.actor_critic = cast(
+                ActorCriticModel, self.config.create_model(**create_model_kwargs),
+            ).to(self.device)
+
+        if initial_model_state_dict is not None:
+            self.actor_critic.load_state_dict(state_dict=initial_model_state_dict)
+        else:
+            assert mode != "train" or self.num_workers == 1, (
```

Comment:
isn't it enough to pass a default seed to all models when more than 1 worker? runner could check whether a seed has been set, and, if not, generate a single random one for all workers. model params should be initialized with the same seed, wouldn't it?